### DISCUSSION TITLE
Execute next CRON job only if last execution finished/failed running

### DISCUSSION THREAD

**Discussion by northcode7** [dataplane-app/dataplane#333] [Q&A] [answered]
Hi Guys,

I think this is an interesting feature. Only execute next trigger scheduler (CRON job) if the last one finished, if it is still running, wait with starting the next one until the last one (current) job finished.

This would be nice to be sure there is always only 1 running instance of your datapipeline.

**Comment by saul-data** [comment, marked as answer]
Makes sense, we can add it to the backlog as a feature.

**Reply by saul-data** [reply]
I gave this a bit of thought and I think it is best that data pipeline handles this and not the CRON. You can do this by inserting a lock record in a database and skip the pipeline if the lock is still present.

**Reply by northcode7** [reply]
A different approach. Sounds good.

**Comment by marconimedeiros2** [comment]
After all, is there a rerun feature or not?
